### ISSUE TITLE
Using compiled_content in postprocess block on Nanoc 4.2.3 generates an error

### ISSUE THREAD

**Issue by werthen** [nanoc/nanoc#913]
Confirmed working on Nanoc 4.2.2, using the method `compiled_content` in a `postprocess` block generates following error:

```
Crashlog created at 2016-07-20 17:43:45 +0200

===== MESSAGE:

Nanoc::Int::Errors::UnmetDependency: The current item cannot be compiled yet because of an unmet dependency on the “/posts/awk.md” item (rep “default”).

===== COMPILATION STACK:

  (empty)

===== STACK TRACE:

  0. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/base/entities/item_rep.rb:92:in `compiled_content'
  1. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/base/views/item_rep_view.rb:46:in `compiled_content'
  2. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/base/views/mixins/with_reps_view_mixin.rb:16:in `compiled_content'
  3. /Users/don/Zeus/zeus.ugent.be/Rules:69:in `block (2 levels) in parse'
  4. /Users/don/Zeus/zeus.ugent.be/Rules:68:in `map'
  5. /Users/don/Zeus/zeus.ugent.be/Rules:68:in `block in parse'
  6. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/rule_dsl/action_provider.rb:63:in `instance_eval'
  7. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/rule_dsl/action_provider.rb:63:in `block in postprocess'
  8. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/rule_dsl/action_provider.rb:62:in `each_value'
  9. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/rule_dsl/action_provider.rb:62:in `postprocess'
  10. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/base/compilation/compiler.rb:78:in `run_all'
  11. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib/nanoc/base/entities/site.rb:30:in `compile'
  12. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:64:in `block in recompile'
  13. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:62:in `chdir'
  14. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:62:in `recompile'
  15. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:54:in `block in recompile_in_subprocess'
  16. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:54:in `fork'
  17. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:54:in `recompile_in_subprocess'
  18. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib/guard/nanoc.rb:19:in `start'
  19. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:82:in `block in _supervise'
  20. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:79:in `catch'
  21. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:79:in `_supervise'
  22. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:22:in `block (3 levels) in run'
  23. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:119:in `block (2 levels) in _run_group_plugins'
  24. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:117:in `each'
  25. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:117:in `block in _run_group_plugins'
  26. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:116:in `catch'
  27. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:116:in `_run_group_plugins'
  28. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:21:in `block (2 levels) in run'
  29. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:20:in `each'
  30. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:20:in `block in run'
  31. /Users/don/.rvm/gems/ruby-2.3.1/gems/lumberjack-1.0.10/lib/lumberjack.rb:32:in `unit_of_work'
  32. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/runner.rb:18:in `run'
  33. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/commander.rb:34:in `start'
  34. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/cli/environments/valid.rb:16:in `start_guard'
  35. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/cli.rb:122:in `start'
  36. /Users/don/.rvm/gems/ruby-2.3.1/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
  37. /Users/don/.rvm/gems/ruby-2.3.1/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
  38. /Users/don/.rvm/gems/ruby-2.3.1/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
  39. /Users/don/.rvm/gems/ruby-2.3.1/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
  40. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/aruba_adapter.rb:32:in `execute'
  41. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib/guard/aruba_adapter.rb:19:in `execute!'

===== VERSION INFORMATION:

Nanoc 4.2.3 © 2007-2016 Denis Defreyne.
Running ruby 2.3.1 (2016-04-26) on x86_64-darwin15 with RubyGems 2.6.6.

===== SYSTEM INFORMATION:

Darwin skye.local 15.5.0 Darwin Kernel Version 15.5.0: Tue Apr 19 18:36:36 PDT 2016; root:xnu-3248.50.21~8/RELEASE_X86_64 x86_64

===== INSTALLED GEMS:

  adsf 1.2.1
  bundler 1.12.5
  coderay 1.1.1
  coffee-script 2.4.1
  coffee-script-source 1.10.0
  colored 1.2
  concurrent-ruby 1.0.2
  cri 2.7.0
  execjs 2.7.0
  ffi 1.9.10
  formatador 0.2.5
  guard 2.14.0
  guard-compat 1.2.1
  guard-nanoc 2.0.0
  hamster 3.0.0
  json 2.0.1
  kramdown 1.10.0
  listen 3.1.5
  lumberjack 1.0.10
  method_source 0.8.2
  mini_portile2 2.1.0
  nanoc 4.2.3
  nenv 0.3.0
  nokogiri 1.6.8
  notiffany 0.1.0
  pkg-config 1.1.7
  pry 0.10.3
  rack 1.6.4
  rb-fsevent 0.9.7
  rb-inotify 0.9.7
  ref 2.0.0
  ruby_dep 1.3.1
  sass 3.4.22
  shellany 0.0.1
  slop 3.6.0
  thor 0.19.1
  w3c_validators 1.2

===== GEMFILE.LOCK:

GIT
  remote: https://github.com/werthen/nanoc.git
  revision: bd3235d8dbae71c5bf0328ba4630070c53a5d3f4
  branch: add-resources-to-ilinks-check
  specs:
    nanoc (4.2.3)
      cri (~> 2.3)
      hamster (~> 3.0)
      ref (~> 2.0)

GEM
  remote: https://rubygems.org/
  specs:
    adsf (1.2.1)
      rack (>= 1.0.0)
    coderay (1.1.1)
    coffee-script (2.4.1)
      coffee-script-source
      execjs
    coffee-script-source (1.10.0)
    colored (1.2)
    concurrent-ruby (1.0.2)
    cri (2.7.0)
      colored (~> 1.2)
    execjs (2.7.0)
    ffi (1.9.10)
    formatador (0.2.5)
    guard (2.14.0)
      formatador (>= 0.2.4)
      listen (>= 2.7, < 4.0)
      lumberjack (~> 1.0)
      nenv (~> 0.1)
      notiffany (~> 0.0)
      pry (>= 0.9.12)
      shellany (~> 0.0)
      thor (>= 0.18.1)
    guard-compat (1.2.1)
    guard-nanoc (2.0.0)
      guard (~> 2.8)
      guard-compat (~> 1.0)
      nanoc (~> 4.0)
    hamster (3.0.0)
      concurrent-ruby (~> 1.0)
    json (2.0.1)
    kramdown (1.10.0)
    listen (3.1.5)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
      ruby_dep (~> 1.2)
    lumberjack (1.0.10)
    method_source (0.8.2)
    mini_portile2 (2.1.0)
    nenv (0.3.0)
    nokogiri (1.6.8)
      mini_portile2 (~> 2.1.0)
      pkg-config (~> 1.1.7)
    notiffany (0.1.0)
      nenv (~> 0.1)
      shellany (~> 0.0)
    pkg-config (1.1.7)
    pry (0.10.3)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    rack (1.6.4)
    rb-fsevent (0.9.7)
    rb-inotify (0.9.7)
      ffi (>= 0.5.0)
    ref (2.0.0)
    ruby_dep (1.3.1)
    sass (3.4.22)
    shellany (0.0.1)
    slop (3.6.0)
    thor (0.19.1)
    w3c_validators (1.2)
      json
      nokogiri

PLATFORMS
  ruby

DEPENDENCIES
  adsf
  coffee-script
  guard-nanoc
  kramdown
  nanoc!
  nokogiri
  sass
  w3c_validators

BUNDLED WITH
   1.12.5

===== LOAD PATHS:

  0. /Users/don/.rvm/gems/ruby-2.3.1/gems/ref-2.0.0/lib
  1. /Users/don/.rvm/gems/ruby-2.3.1/gems/bundler-1.12.5/lib
  2. /Users/don/.rvm/gems/ruby-2.3.1/gems/w3c_validators-1.2/lib
  3. /Users/don/.rvm/gems/ruby-2.3.1/gems/sass-3.4.22/lib
  4. /Users/don/.rvm/gems/ruby-2.3.1@global/gems/nokogiri-1.6.8/lib
  5. /Users/don/.rvm/gems/ruby-2.3.1@global/extensions/x86_64-darwin-15/2.3.0/nokogiri-1.6.8
  6. /Users/don/.rvm/gems/ruby-2.3.1@global/gems/pkg-config-1.1.7/lib
  7. /Users/don/.rvm/gems/ruby-2.3.1@global/gems/mini_portile2-2.1.0/lib
  8. /Users/don/.rvm/gems/ruby-2.3.1/gems/kramdown-1.10.0/lib
  9. /Users/don/.rvm/gems/ruby-2.3.1/gems/json-2.0.1/lib
  10. /Users/don/.rvm/gems/ruby-2.3.1/extensions/x86_64-darwin-15/2.3.0/json-2.0.1
  11. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-nanoc-2.0.0/lib
  12. /Users/don/.rvm/gems/ruby-2.3.1/bundler/gems/nanoc-bd3235d8dbae/lib
  13. /Users/don/.rvm/gems/ruby-2.3.1/gems/ref-2.0.0/lib
  14. /Users/don/.rvm/gems/ruby-2.3.1/gems/hamster-3.0.0/lib
  15. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-compat-1.2.1/lib
  16. /Users/don/.rvm/gems/ruby-2.3.1/gems/guard-2.14.0/lib
  17. /Users/don/.rvm/gems/ruby-2.3.1/gems/thor-0.19.1/lib
  18. /Users/don/.rvm/gems/ruby-2.3.1/gems/pry-0.10.3/lib
  19. /Users/don/.rvm/gems/ruby-2.3.1/gems/slop-3.6.0/lib
  20. /Users/don/.rvm/gems/ruby-2.3.1/gems/method_source-0.8.2/lib
  21. /Users/don/.rvm/gems/ruby-2.3.1/gems/notiffany-0.1.0/lib
  22. /Users/don/.rvm/gems/ruby-2.3.1/gems/shellany-0.0.1/lib
  23. /Users/don/.rvm/gems/ruby-2.3.1/gems/nenv-0.3.0/lib
  24. /Users/don/.rvm/gems/ruby-2.3.1/gems/lumberjack-1.0.10/lib
  25. /Users/don/.rvm/gems/ruby-2.3.1/gems/listen-3.1.5/lib
  26. /Users/don/.rvm/gems/ruby-2.3.1/gems/ruby_dep-1.3.1/lib
  27. /Users/don/.rvm/gems/ruby-2.3.1/gems/rb-inotify-0.9.7/lib
  28. /Users/don/.rvm/gems/ruby-2.3.1/gems/rb-fsevent-0.9.7/lib
  29. /Users/don/.rvm/gems/ruby-2.3.1/gems/formatador-0.2.5/lib
  30. /Users/don/.rvm/gems/ruby-2.3.1/gems/ffi-1.9.10/lib
  31. /Users/don/.rvm/gems/ruby-2.3.1/extensions/x86_64-darwin-15/2.3.0/ffi-1.9.10
  32. /Users/don/.rvm/gems/ruby-2.3.1/gems/cri-2.7.0/lib
  33. /Users/don/.rvm/gems/ruby-2.3.1@global/gems/concurrent-ruby-1.0.2/lib
  34. /Users/don/.rvm/gems/ruby-2.3.1/gems/colored-1.2/lib
  35. /Users/don/.rvm/gems/ruby-2.3.1/gems/coffee-script-2.4.1/lib
  36. /Users/don/.rvm/gems/ruby-2.3.1/gems/execjs-2.7.0/lib
  37. /Users/don/.rvm/gems/ruby-2.3.1/gems/coffee-script-source-1.10.0/lib
  38. /Users/don/.rvm/gems/ruby-2.3.1/gems/coderay-1.1.1/lib
  39. /Users/don/.rvm/gems/ruby-2.3.1/gems/adsf-1.2.1/lib
  40. /Users/don/.rvm/gems/ruby-2.3.1/gems/rack-1.6.4/lib
  41. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/site_ruby/2.3.0
  42. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/site_ruby/2.3.0/x86_64-darwin15
  43. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/site_ruby
  44. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/vendor_ruby/2.3.0
  45. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin15
  46. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/vendor_ruby
  47. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0
  48. /Users/don/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/x86_64-darwin15
```

**Comment by ddfreyne** [comment]
This is most likely an unintended consequence of #894.

**Comment by ddfreyne** [comment]
Fixed in #917.
